src/devices/Apa102/Apa102.cs

```diff
@@ -28,8 +28,7 @@ public class Apa102 : IDisposable
         /// <param name="length">Number of LEDs</param>
         public Apa102(SpiDevice spi, int length)
         {
-            _spi = spi ?? throw new ArgumentNullException(nameof(spi));
```

Comment:
we don't do validation anymore? I thought the [guidelines](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/api-guidelines/nullability.md#annotation-guidance) said to retain validation:

```
DO NOT remove existing argument validation when annotating existing public/protected APIs. If through the course of annotating it becomes clear that internal/private surface area is unnecessarily checking for nulls when nulls aren't possible (i.e. you found dead code), such use can be removed or replaced by asserts.
```

Reply:
(I'm fine either way but would be weird that if people didn't opt in for the NRT feature they would start getting NRE because of the feature which was supposed to prevent them from getting it)

Reply:
please keep the validation, one good reason: nanoFramework does not support nullable.

Reply:
I wasn't sure about that. Yes, we should follow the guidelines.

Reply:
I fixed this. I also found other cases like this and restored them.